Sources/Tools/MaxMain/MaxCompat.h

```diff
@@ -78,21 +96,83 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 #   define SetParamBlock2Controller(pb, id, tab, ctl) pb->SetControllerByID(id, tab, ctl)
 #endif // MAX_VERSION_MAJOR
 
-#if MAX_VERSION_MAJOR <= 11 // max 2009. Just a guess, really. 2010 doesn't need this function.
-#   define INIT_CUSTOM_CONTROLS(instance) InitCustomControls(instance)
+#if MAX_VERSION_MAJOR <= 14 // Max 2012
+#   define p_end end
+#   define MAX14_CONST
+#   define GETDLGTEXT_RETURN_TYPE TCHAR*
+#   define READ_VOID_BUFFER(s) s->Read
+#   define WRITE_VOID_BUFFER(s) s->Write
+#   define BM_LOAD_CONFIGURE_PARAMS void* ptr
+#   define BMNAME_VALUE_TYPE TCHAR*
 #else
-#   define INIT_CUSTOM_CONTROLS(instance)
+#   define MAX14_CONST const
+#   define GETDLGTEXT_RETURN_TYPE const MCHAR*
+#   define READ_VOID_BUFFER(s) s->ReadVoid
+#   define WRITE_VOID_BUFFER(s) s->WriteVoid
+#   define BM_LOAD_CONFIGURE_PARAMS void* ptr, DWORD piDataSize
```

Comment:
This should probably just be a single parameter macro like the `MAX_REF_PROPAGATE` below